velox/functions/sparksql/Hash.cpp

```diff
@@ -157,6 +162,24 @@ class PrimitiveVectorHasher : public SparkVectorHasher<HashClass> {
   }
 };
 
+template <typename HashClass>
+class UnknowTypeVectorHasher : public SparkVectorHasher<HashClass> {
+ public:
+  using SeedType = typename HashClass::SeedType;
+  using ReturnType = typename HashClass::ReturnType;
+
+  UnknowTypeVectorHasher(DecodedVector& decoded)
+      : SparkVectorHasher<HashClass>(decoded) {}
+
+  ReturnType hashAt(vector_size_t index, SeedType seed) override {
+    return seed;
+  }
+
+  ReturnType hashNotNullAt(vector_size_t index, SeedType seed) override {
+    VELOX_FAIL("hashNotNullAt should not be called for an unknown type");
```

Comment:
nit: use VELOX_USER_FAIL as it is an error caused by the user.
message suggestion: remove `an` and add `.` at the end.

Reply:
This should be a server side error since if hint here, must be caller mistakenly call `UnknowTypeVectorHasher. hashNotNullAt`, `UnknowTypeVectorHasher` should only be called `hashAt`